src/compiler/checker.ts

```diff
@@ -32023,14 +32045,48 @@ namespace ts {
         }
 
         function checkInExpression(left: Expression, right: Expression, leftType: Type, rightType: Type): Type {
-            if (leftType === silentNeverType || rightType === silentNeverType) {
-                return silentNeverType;
+            if (isPrivateIdentifier(left)) {
+                const privateId = left;

```

Comment:
Is it necessary to alias `left` as `privateId`? `left` will have the type `PrivateIdentifier` throughout the body of the block, so it doesn't seem necessary.

Reply:
agreed, the function is small enough to not lose context of what left is.